langchain4j-weaviate/src/main/java/dev/langchain4j/store/embedding/WeaviateEmbeddingStoreImpl.java

```diff
@@ -0,0 +1,192 @@
+import io.weaviate.client.v1.graphql.query.argument.NearVectorArgument;
+import io.weaviate.client.v1.graphql.query.fields.Field;
+import java.security.MessageDigest;
+import java.util.*;
+import java.util.stream.Collectors;
+import lombok.Builder;
+import lombok.SneakyThrows;
+import lombok.val;
+
+public class WeaviateEmbeddingStoreImpl implements EmbeddingStore<TextSegment> {
+
+  private static final String DEFAULT_CLASS = "Default";
+  private static final Double DEFAULT_MIN_CERTAINTY = 0.8;
+  private static final String METADATA_TEXT_SEGMENT = "text";
+  private static final String ADDITIONALS = "_additional";
+
+  private final WeaviateClient client;
+  private final String objectClass;
+
+  @Builder
+  @SneakyThrows
+  public WeaviateEmbeddingStoreImpl(String apiKey, String scheme, String host, String objectClass) {
+    client = WeaviateAuthClient.apiKey(new Config(scheme, host), apiKey);
+    this.objectClass = objectClass != null ? objectClass : DEFAULT_CLASS;
+  }
+
+  @Override
+  public String add(Embedding embedding) {
+    val id = generateRandomId();
+    add(id, embedding);
+    return id;
+  }
+
+  @Override
+  public void add(String id, Embedding embedding) {
+    addAll(singletonList(id), singletonList(embedding), null);
+  }
+
+  @Override
+  public String add(Embedding embedding, TextSegment textSegment) {
+    return addAll(singletonList(embedding), textSegment == null ? null : singletonList(textSegment))
+      .stream()
+      .findFirst()
+      .orElse(null);
+  }
+
+  @Override
+  public List<String> addAll(List<Embedding> embeddings) {
+    return addAll(embeddings, null);
+  }
+
+  @Override
+  public List<String> addAll(List<Embedding> embeddings, List<TextSegment> embedded) {
+    return addAll(null, embeddings, embedded);
+  }
+
+  @Override
+  public List<EmbeddingMatch<TextSegment>> findRelevant(Embedding referenceEmbedding, int maxResults) {
+    return findRelevant(referenceEmbedding, maxResults, DEFAULT_MIN_CERTAINTY);
+  }
+
+  @Override
+  public List<EmbeddingMatch<TextSegment>> findRelevant(
+    Embedding referenceEmbedding,
+    int maxResults,
+    double minCertainty
+  ) {
+    val result = client
+      .graphQL()
+      .get()
+      .withClassName(objectClass)
+      .withFields(
+        Field.builder().name(METADATA_TEXT_SEGMENT).build(),
+        Field
+          .builder()
+          .name(ADDITIONALS)
+          .fields(
+            Field.builder().name("id").build(),
+            Field.builder().name("certainty").build(),
+            Field.builder().name("vector").build()
+          )
+          .build()
+      )
+      .withNearVector(
+        NearVectorArgument
+          .builder()
+          .vector(referenceEmbedding.vectorAsList().toArray(new Float[0]))
+          .certainty((float) minCertainty)
+          .build()
+      )
+      .withLimit(maxResults)
+      .run();
+
+    if (result.hasErrors()) {
+      return emptyList();
```

Comment:
Should we throw an exception with details in this case?

Reply:
BTW when I tried `Weaviate_Vector_Database_Example `and put a wrong host by mistake (included "http://" in it), I did not get any warnings or errors, it (not) worked silently and pretended to be answering questions. But under the hood weaviate was not working I guess. Is it possible to change it to fail fast in such cases?

Reply:
I've just added throwing exception in `hasErrors` case. If wrong host is still accepted, then I would leave it to be solved on the nested `io.weaviate` lib, right?

Reply:
schema and host remain split, since it is so in nested `io.weaviate` lib, is fine for you?

Reply:
regarding "https://" inside "host", it is now failing after your changes :+1: :
```
Exception in thread "main" java.lang.IllegalArgumentException: https
	at dev.langchain4j.store.embedding.WeaviateEmbeddingStoreImpl.findRelevant(WeaviateEmbeddingStoreImpl.java:124)
	at dev.langchain4j.store.embedding.WeaviateEmbeddingStoreImpl.findRelevant(WeaviateEmbeddingStoreImpl.java:87)
	at dev.langchain4j.store.embedding.WeaviateEmbeddingStore.findRelevant(WeaviateEmbeddingStore.java:133)
	at dev.langchain4j.retriever.EmbeddingStoreRetriever.findRelevant(EmbeddingStoreRetriever.java:37)
	at dev.langchain4j.chain.ConversationalRetrievalChain.execute(ConversationalRetrievalChain.java:55)
	at ChatWithDocumentsExamples$Weaviate_Vector_Database_Example.main(ChatWithDocumentsExamples.java:228)
```

Reply:
read again - schema and host remain split, so why "https://" inside "host"? 😄 there is no change

Reply:
When one copies URL from the weaviate dashboard, it includes schema as well. When I first did it, I did not pay attention to this and pasted it with "https://" into "host" in the example you created. When I first run it, it was not working, but there was no sign of failures (no exceptions, nothing) and it was very confusing... This is why I asked you to throw an exception in case of errors. After your changes it throws an exception and it is clear to the user that comething is wrong.